demo/utils.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+from itertools import count
+import os
+
+API = "https://api.github.com"
+
+# Get a list of repository names by username
+def get_all_repositories(username):
+    repo_names = []
+    for page in count(1):
+        repo_names_by_page = get_repositories_by_page(page, username)
+        if not repo_names_by_page:
+            break
+        repo_names.extend(repo_names_by_page)
+    return repo_names
+
+def get_repositories_by_page(page, username):
+    URL = API + "/users/%s/repos" % (username)
+    query_parameters = {'page': page}
+    response = requests.get(url=URL, params=query_parameters)
```

Comment:
nit:
`
    response = requests.get(url=URL,headers=headers)
    # handle error code and status
    json_response = response.json()
    return json_response
`
seems to be a very common pattern. Sharing error handling code can also increase the debuggability of the code. I think it's worth creating a function for it.

Reply:
Thanks! Will fix it.